packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -0,0 +1,549 @@
+    }
+
+    if (typeof this.props.onDragEnd === 'function') {
+      this.props.onDragEnd(event);
+    }
+  };
+
+  handleMouseMove = event => {
+    const { min, max, vertical, reverse } = this.props;
+    const percent = calculatePercent(this.container, event, vertical, reverse);
+    const value = percentToValue(percent, min, max);
+
+    this.emitChange(event, value);
+  };
+
+  playJumpAnimation() {
+    this.setState({ currentState: 'jumped' }, () => {
+      setTimeout(() => {
+        this.setState({ currentState: 'normal' });
+      }, this.props.theme.transitions.duration.complex);
+    });
+  }
+
+  render() {
+    const { currentState } = this.state;
+    const {
+      component: Component,
+      classes,
+      value,
+      min,
+      max,
+      vertical,
+      reverse,
+      disabled,
+    } = this.props;
+
+    const percent = clamp((value - min) * 100 / (max - min));
+
+    const commonClasses = {
+      [classes.disabled]: disabled,
+      [classes.jumped]: !disabled && currentState === 'jumped',
+      [classes.focused]: !disabled && currentState === 'focused',
+      [classes.activated]: !disabled && currentState === 'activated',
+    };
+
+    const containerClasses = classNames(classes.container, {
+      [classes.vertical]: vertical,
+      [classes.reverse]: reverse,
+      [classes.disabled]: disabled,
+    });
+
+    const trackBeforeClasses = classNames(classes.track, classes.trackBefore, commonClasses, {
+      [classes.vertical]: vertical,
+    });
+
+    const trackAfterClasses = classNames(classes.track, classes.trackAfter, commonClasses, {
+      [classes.vertical]: vertical,
+    });
+
+    const thumbClasses = classNames(classes.thumb, commonClasses, {
+      [classes.zero]: percent === 0,
+    });
+
+    const trackProperty = vertical ? 'height' : 'width';
+    const thumbProperty = vertical ? 'top' : 'left';
+    const inlineTrackBeforeStyles = { [trackProperty]: this.calculateTrackBeforeStyles(percent) };
+    const inlineTrackAfterStyles = { [trackProperty]: this.calculateTrackAfterStyles(percent) };
+    const inlineThumbStyles = { [thumbProperty]: `${percent}%` };
+
+    const ariaProps = getAriaProps(this.props);
+
+    return (
+      <Component
+        role="slider"
+        className={containerClasses}
+        aria-valuenow={value}
+        aria-valuemin={min}
+        aria-valuemax={max}
+        aria-orientation={vertical ? 'vertical' : 'horizontal'}
+        onClick={this.handleClick}
+        ref={node => {
+          this.container = findDOMNode(node);
+        }}
+        {...ariaProps}
+      >
+        <div className={trackBeforeClasses} style={inlineTrackBeforeStyles} />
+        <ButtonBase
+          className={thumbClasses}
+          disableRipple
+          style={inlineThumbStyles}
+          onBlur={this.handleBlur}
+          onKeyDown={this.handleKeyDown}
+          onMouseDown={this.handleMouseDown}
+          onTouchStartCapture={this.handleTouchStart}
```

Comment:
Why do we need to trigger the event during the capture phase? It's the first time we do such.

Reply:
I need to prevent page scrolling when moving the slider (function preventPageScrolling).I should catch event before preventing, so I use capture phase.